mitmproxy/addons/session.py

```diff
@@ -40,6 +73,16 @@ def __del__(self):
         if self.tempdir:
             shutil.rmtree(self.tempdir)
 
+    def __contains__(self, fid):
```

Comment:
I don't like the use of `__contains__` here. This is an expensive operation, which goes to disk and retrieves IDs of all rows. Being able to use `if x in db` encourages casual use of this operation (see comments below) that we should actually use carefully. 

We need to either make this operation cheap (keep an in-memory set of all IDs on disk or at least make it only retrieve one record), or make it a function like "has_id" so the user is a bit more prepared. I think having an in-memory set of IDs should be fine if this is an important operation.